test/run

```diff
@@ -159,13 +159,22 @@ function test_from_dockerfile_s2i() {
 }
 
 app_cleanup() {
+  info "Cleaning up the test app folders"
   for server in ${WEB_SERVERS[*]}; do
-    if [ ! -z "${server}" ]; then
+    if [[ "${server}" == "db" ]]; then
+      echo "Removing ${test_dir}/db-test-app"
```

Comment:
This change is indeed correct as the condition makes only sense in the look. On the other hand, if we run `git clone` no matter what `WEB_SERVERS` includes, shouldn't we remove the `db-test-app` also no matter what the  `WEB_SERVERS` variable includes (i.e. leave it outside of the loop and even without the condition)?

Reply:
Another thing is that it would be better to clone the db-test-app repository that we want to remove at the end of tests into a different path then the sources -- somewhere under `/tmp/XYZ/db-test-app` would likely be a better place to not spoil the source dir with temporary+test-only files. But that also might be a different PR, if you don't want to complicate this one.

Reply:
AFAIK, the `db-test-app` folder gets created only when testing the 'db'. This happens if `WEB_SERVERS` variable contain the 'db' string, but otherwise it shouldn't be tested, nor the folder created.

Maybe I'm missing something here, so please correct me if I'm wrong, but I feel like adding this to the loop were supposed to prevent needless calls to `rm -rf` on folders that do not exist.

Reply:
Ack for the `/tmp` part, let's ignore it for this PR.

> AFAIK, the `db-test-app` folder gets created only when testing the 'db'.

That's what IMO should happen, but it is not -- see https://github.com/sclorg/s2i-ruby-container/blob/f07ad29251018bdea712227561653f2c47c82771/test/run#L220, the `git clone` is run every-time the `run` script is called. That's why I suggested to remove the `db-test-app` folder every-time as well.

Or, alternatively, we can fix the code to behave as you say -- only clone the git when `db` use case is tested, which can be done either by adding a condition around `git clone` or by moving the clone to inside the loop.

Reply:
> That's what IMO should happen, but it is not -- see
oh...

Well, both solutions here sound a little hacky, because this whole process will only be used for this one specific test. I feel like simply cloning every time and deleting every time is more simple, it will be happening in the current state always anyway. I'd want to steer away from complicating things more than they need to be. Every condition is another thing to debug. We can think about the git repository as a _current_ dependency for tests, if we remove the db test we can remove that dependency.

In this sense, the most clean approach would be to git clone before all tests (like we do now) and `rm -rf` in the same scope.

Ideally there could be some kind of dependency system for test, that could download everything before starting tests and delete after, but that would be overengineering it for one single git repo here.

Reply:
> In this sense, the most clean approach would be to git clone before all tests (like we do now) and `rm -rf` in the same scope.

Sounds good to me.
 
> Ideally there could be some kind of dependency system for test, that could download everything before starting tests and delete after, but that would be overengineering it for one single git repo here.

+1

Reply:
Cleanup further modified based on this discussion. Rewrote commit message to reflect this discussion.

Included comments around cleaning in bash script to make it as obvious as possible that the `git clone` and `rm -rf` are related.

Reply:
Thank you, @hhorak, for the review and the discussion, it was very helpful.